### ISSUE TITLE
repeat loading lsp layer when enable lsp for two or more language layers

### ISSUE THREAD

**Issue by sunlin7** [syl20bnr/spacemacs#15674]
Enable the `lsp` and `javascript` and `c-c++` layers in dot-spacemacs, then start emacs with debug init option, checking the log in `*Message*` buffer, can find the `lsp/layers.el` is loaded more than once ( if enable more lsp-dependened layers, the lsp layer will be loaded again and again ). 
>$ emacs --debug-init
layers/+lang/c-c++/layers.el (source)...
layers/+tools/lsp/layers.el (source)...done
layers/+tools/dap/layers.el (source)...
layers/+tools/lsp/layers.el (source)...done
layers/+tools/dap/layers.el (source)...done
layers/+lang/c-c++/layers.el (source)...done
layers/+tools/dap/packages.el (source)...done
layers/+tools/dap/layers.el (source)...
layers/+tools/lsp/layers.el (source)...done
layers/+tools/dap/layers.el (source)...done
layers/+lang/javascript/packages.el (source)...done
layers/+lang/javascript/layers.el (source)...
layers/+lang/json/layers.el (source)...done
layers/+tools/lsp/layers.el (source)...done
layers/+lang/javascript/layers.el (source)...done
layers/+tools/lsp/packages.el (source)...done
layers/+tools/lsp/layers.el (source)...done
layers/+lang/typescript/packages.el (source)...done
layers/+lang/typescript/layers.el (source)...
layers/+lang/javascript/layers.el (source)...
layers/+lang/json/layers.el (source)...done
layers/+tools/lsp/layers.el (source)...done
layers/+lang/javascript/layers.el (source)...done
layers/+lang/typescript/layers.el (source)...done

**Comment by lebensterben** [comment]
this looks like a misconfiguration on your side.

when starting emacs, lsp or any language layers should not be loaded at all, until they are used.

**Comment by lebensterben** [comment]
how to reproduce this?

When emacs is launched with --debug-init, I don't see any of these loaded.

**Comment by sunlin7** [comment]
Hi @lebensterben just enable the `lsp` layer and `c-c++` layers with lsp backend, for example, enable follow layers will be able to reproduce the issue.
``` emacs-lisp
lsp
(c-c++ :variables c-c++-backend 'lsp-clangd)
(javascript :variables javascript-backend 'lsp)
```

And according the traceback, the function `configuration-layer/declare-layer-dependencies` will be called from the `layers/+lang/c-c++/layers.el` and `layers/+lang/javascript/layers.el`, which will call the `lsp/layers` again and again.

**Comment by sunlin7** [comment]
Close for resolved by #15772.
